src/v0/destinations/twitter_ads/transform.js

```diff
@@ -0,0 +1,185 @@
+const sha256 = require('sha256');
+
+const {
+  constructPayload,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  isDefinedAndNotNull,
+  simpleProcessRouterDest,
+} = require('../../util');
+const { EventType } = require('../../../constants');
+const {
+  ConfigCategories,
+  mappingConfig,
+  BASE_URL
+} = require('./config');
+
+const { InstrumentationError, OAuthSecretError, ConfigurationError } = require('../../util/errorTypes');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+const { getAuthHeaderForRequest } = require("./util");
+
+const getOAuthFields = ({ secret }) => {
+  if (!secret) {
+    throw new OAuthSecretError('[TWITTER ADS]:: OAuth - access keys not found');
+  }
+  const oAuthObject = {
+    consumerKey: secret.consumerKey,
+    consumerSecret: secret.consumerSecret,
+    accessToken: secret.accessToken,
+    accessTokenSecret: secret.accessTokenSecret
+  };
+  return oAuthObject;
+};
+
+// build final response
+function buildResponse(message, requestJson, metadata, endpointUrl) {
+  const response = defaultRequestConfig();
+  response.endpoint = endpointUrl;
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.body.JSON.conversions = [removeUndefinedAndNullValues(requestJson)];
+  // required to be in accordance with oauth package
+  const request = {
+    url: response.endpoint,
+    method: response.method,
+    body: response.body.JSON
+  };
+
+  const oAuthObject = getOAuthFields(metadata);
+  const authHeader = getAuthHeaderForRequest(request, oAuthObject).Authorization;
+  response.headers = {
+    Authorization: authHeader,
+    'Content-Type': JSON_MIME_TYPE,
+  };
+  return response;
+}
+
+function prepareUrl(message, destination) {
+  const pixelId = message.properties.pixelId || destination.Config.pixelId;
+  return `${BASE_URL}/${pixelId}`;
+}
+
+function populateEventId(event, requestJson, destination) {
+
+  const eventNameToIdMappings = destination.Config.twitterAdsEventNames;
+  let eventId = "";
+
+  if (eventNameToIdMappings) {
+    eventNameToIdMappings.some(obj => {
```

Comment:
correct method will be [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find), not some

Reply:
updated